.github/workflows/build_test_upload.yml

```diff
@@ -90,6 +93,52 @@ jobs:
           MATCH_KEYCHAIN_NAME: ${{ secrets.MATCH_KEYCHAIN_NAME }}
           MATCH_KEYCHAIN_PASSWORD: ${{ secrets.MATCH_KEYCHAIN_PASSWORD }}
           SOURCE_BRANCH: ${{ github.head_ref }}
+      - name: Install code coverage tool
+        run: |
+          brew tap a7ex/homebrew-formulae
+          brew install xcresultparser
+      - name: Create cobertura code coverage file
+        run: |
+          ls ./fastlane/test_output/
+          xcresultparser -o cobertura ./fastlane/test_output/PrimerSDKTests.xcresult > cobertura.xml
+          xcresultparser -o md ./fastlane/test_output/PrimerSDKTests.xcresult > results.md
+      - uses: actions/upload-artifact@master
+        with:
+          name: cobertura-coverage-file
+          path: ./cobertura.xml
+      - uses: actions/upload-artifact@master
+        with:
+          name: results-markdown
+          path: ./results.md
+
+  test-coverage:
+    needs: 
+      - sdk-unit-tests
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - uses: actions/download-artifact@master
+        with:
+          name: cobertura-coverage-file
+          path: ./cobertura.xml
+      - name: Code Coverage Summary
+        uses: irongut/CodeCoverageSummary@v1.3.0
+        with: 
+          filename: "*/cobertura.xml"
+          badge: true
+          format: markdown
+          hide_branch_rate: false
+          indicators: true
+          fail_below_min: true
+          thresholds: '25 75'
+          output: both
+      - name: Add Coverage PR Comment
+        uses: marocchino/sticky-pull-request-comment@v2
```

Comment:
Lines 105, 109, 120, 125 we will need to use a sha commits as per security guidelines